pkg/util/error/distributor.go

```diff
@@ -0,0 +1,129 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+
+package error
+
+import "github.com/grafana/mimir/pkg/util/validation"
+
+type DistributorError interface {
+	error
+	DistributorError()
+}
+
+// DistributorPushError is a generic error returned on
+// distributor's write path.
+type DistributorPushError struct {
+	err error
+}
+
+// Error makes DistributorPushError implement error interface.
+func (e DistributorPushError) Error() string {
+	return e.err.Error()
+}
+
+// Unwrap makes DistributorPushError implement Wrapper interface.
+func (e DistributorPushError) Unwrap() error {
+	return e.err
+}
+
+// PushError makes DistributorPushError implement pusherror.Error interface.
+func (e DistributorPushError) PushError() {
+}
+
+// DistributorError makes DistributorPushError implement DistributorError interface.
+func (e DistributorPushError) DistributorError() {
+}
```

Comment:
Style nit, I think it could be opinonated for one liners above, but for empty methods I think that the newline removes readability:

```suggestion
// PushError makes DistributorPushError implement pusherror.Error interface.
func (e DistributorPushError) PushError() {}

// DistributorError makes DistributorPushError implement DistributorError interface.
func (e DistributorPushError) DistributorError() {}
```